src/Tizen.Applications.Common/Tizen.Applications/LocaleManager.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Tizen.Applications
+{
+    static class LocaleManager
+    {
+        private static readonly string LogTag = "Tizen.Applications";
+        private static readonly string SupportedLocalesFilePath = "/usr/share/i18n/SUPPORTED";
+        private static bool _fileExists = File.Exists(SupportedLocalesFilePath);
+        private static HashSet<string> _supportedLocales;
+
+        public static void SetApplicationLocale(CultureInfo info)
+        {
+            if (!_fileExists)
+            {
+                Log.Error(LogTag, $"{SupportedLocalesFilePath} not found");
+                return;
+            }
+            var converter = new SystemLocaleConverter();
+            string convertedLocale = converter.Convert(info);
+
+            if (VerifySupportedLocale(convertedLocale))
+            {
+                Interop.AppCommon.AppCommonErrorCode ret = Interop.AppCommon.AppLocaleManagerSetLanguage(convertedLocale);
+                if (ret != Interop.AppCommon.AppCommonErrorCode.None)
+                {
+                    Interop.AppCommon.AppLocaleManagerSetLanguage("en_US.UTF-8");
+                }
+            }
+            else
+            {
+                Log.Error(LogTag, $"Unsupported Language : {info.Name}");
+            }
+        }
+
+        public static CultureInfo GetApplicationLocale()
+        {
+            IntPtr langPtr = IntPtr.Zero;
+
+            try
+            {
+                Interop.AppCommon.AppCommonErrorCode ret = Interop.AppCommon.AppLocaleManagerGetLanguage(out langPtr);
+
+                if (ret != Interop.AppCommon.AppCommonErrorCode.None || langPtr == IntPtr.Zero)
+                {
+                    Log.Error(LogTag, $"Failed to get application language. Result: {ret}");
+                    return null;
+                }
+
+                string language = Marshal.PtrToStringAnsi(langPtr);
+                if (string.IsNullOrEmpty(language))
+                {
+                    Log.Warn(LogTag, "Retrieved application language is null or empty.");
+                    return null;
+                }
+
+                var converter = new SystemLocaleConverter();
+                return converter.Convert(language);
+            }
+            finally
+            {
+                if (langPtr != IntPtr.Zero)
+                {
+                    Marshal.FreeCoTaskMem(langPtr);
+                }
+            }
+        }
+
+        public static string GetSystemLocale()
```

Comment:
The return value should be 'CultureInfo'.

Reply:
Done :) [commit](https://github.com/Samsung/TizenFX/pull/7174/commits/d0f4f9fc617f71b2c614bf44c12cf2969bdb822b)